examples/chat/common/src/lib.rs

```diff
@@ -14,4 +14,39 @@
 // limitations under the License.
 //
 
+pub mod msg;
 pub mod proto;
+
+use anyhow::{anyhow, Context, Result};
+
+/// A trait for objects that can be encoded as bytes + handles.
+pub trait Encodable {
+    fn encode(&self) -> Result<(Vec<u8>, Vec<oak::Handle>)>;
+}
+
+/// A trait for objects that can be decoded from bytes + handles.
```

Comment:
This is nice &ndash; I think it should live under `sdk/rust/oak/` somewhere for more general use (but that can be a separate PR).